classifier/classifier.py

```diff
@@ -1,138 +1,247 @@
+
+        self.parser.add_argument("-o", "--output", type=str,
+                                 help="Main directory to put organized folders")
+
+        self.parser.add_argument("-d", "--directory", type=str,
+                                 help="The directory whose files to classify")
+
+        self.parser.add_argument("-dt", "--date", action='store_true',
+                                 help="Organize files by creation date")
+
+        self.parser.add_argument("-df", "--dateformat", type=str,
+                                 help="set the date format using YYYY, MM or DD")
+
+        self.args = self.parser.parse_args()
+        self.dateformat = 'YYYY-MM-DD'
+        self.formats = {}
+        self.dirconf = None
+        self.checkconfig()
+        self.main()
+
+    def checkconfig(self):
+        """ create a default config if not available """
+        if not os.path.isdir(os.path.dirname(CONFIG)):
+            os.makedirs(os.path.dirname(CONFIG))
+        if not os.path.isfile(CONFIG):
+            conffile = open(CONFIG, "w")
+            conffile.write("IGNORE:.part,.desktop\n" +
+                           "Music:.mp3,.aac,.flac,.ogg,.wma,.m4a,.aiff,.wav,.amr\n" +
+                           "Videos:.flv,.ogv,.avi,.mp4,.mpg,.mpeg,.3gp,.mkv,.ts,.webm,.vob,.wmv\n" +
+                           "Pictures:.png,.jpeg,.gif,.jpg,.bmp,.svg,.webp,.psd,.tiff\n" +
+                           "Archives:.rar,.zip,.7z,.gz,.bz2,.tar,.dmg,.tgz,.xz,.iso,.cpio\n" +
+                           "Documents:.txt,.pdf,.doc,.docx,.odf,.xls,.xlsv,.xlsx," +
+                           ".ppt,.pptx,.ppsx,.odp,.odt,.ods,.md,.json,.csv\n" +
+                           "Books:.mobi,.epub,.chm\n" +
+                           "DEBPackages:.deb\n" +
+                           "Programs:.exe,.msi\n" +
+                           "RPMPackages:.rpm")
+            conffile.close()
+        for items in open(CONFIG, "r"):
+            (key, val) = items.split(':')
+            self.formats[key] = val
+        return
+
+    def moveto(self, filename, from_folder, to_folder):
+        from_file = os.path.join(from_folder, filename)
+        to_file = os.path.join(to_folder, filename)
+        # to move only files, not folders
+        if not to_file == from_file:
+            print('  moving:', to_file)
+            if os.path.isfile(from_file):
+                if not os.path.exists(to_folder):
+                    os.makedirs(to_folder)
+                os.rename(from_file, to_file)
+        return
+
+    def classify(self, formats, output, directory):
+        for file in os.listdir(directory):
+            tmpbreak = False
+            # set up a config per folder
+            if not file == DIRCONFFILE and os.path.isfile(os.path.join(directory, file)):
+                filename, file_ext = os.path.splitext(file)
+                file_ext = file_ext.lower()
+                if len(self.formats) > 1:
+                    if self.formats['IGNORE']:
```

Comment:
Not the way to check if a key exists in a dictionary. It will throw KeyError.
Instead use:
```python
if 'IGNORE' in self.formats:
```